pydecorate/decorator_base.py

```diff
@@ -462,134 +442,287 @@ def _add_scale(self, colormap, title=None, **kwargs):
 
         # reload draw object
         draw = self._get_canvas(self.image)
+        self._draw_colorbar_ticks(
+            draw,
+            minval,
+            maxval,
+            is_vertical,
+            scale_width,
+            scale_height,
+            px,
+            py,
+            mx,
+            my,
+            x,
+            y,
+        )
+
+        # draw unit and/or power if set
+        if self.style["unit"]:
+            self._add_colorbar_units(
+                draw,
+                self.style["unit"],
+                is_vertical,
+                is_right,
+                is_bottom,
+                x,
+                y,
+                mx,
+                my,
+                scale_width,
+                scale_height,
+                x_spacer,
+                y_spacer,
+            )
+
+        if title:
+            self._add_colorbar_title(
+                draw,
+                title,
+                is_vertical,
+                is_right,
+                is_bottom,
```

Comment:
Yes. The next step further down the refactor rabbit hole would be to put all the colorbar stuff in the method into one helper class. It would be much easier to access all the size properties that it uses. I don't know if I want to go down that hole though in this PR. One reason I don't want to is the weird design of pydecorate with its `self.style` dictionary that is read from and written to in just about every method. It is very not...good.

Reply:
Yes we should work on this some day.